### PR TITLE
[featuregate] Automatically set GOMEMLIMIT and GOMAXPROCS for collector, target allocator, opamp bridge

### LINKED ISSUE #1456: [Bug]: otelcol can't get cpu limit and may cause performance issues
### What happened?

See
- https://github.com/open-telemetry/opentelemetry-collector/issues/4459

### Steps to reproduce

Details are in the collector issue

### Expected behavior

detecting the correct amount of cpus

### Relevant log output

_No response_

### Screenshot

_No response_

### Additional context

_No response_

### Jaeger backend version

_No response_

### SDK

_No response_

### Pipeline

_No response_

### Stogage backend

_No response_

### Operating system

_No response_

### Deployment model

_No response_

### Deployment configs

_No response_

### LINKED ISSUE #2919: Automatically set GOMEMLIMIT for collector, TA, bridge, etc.
### Component(s)

collector, target allocator, opamp

### Is your feature request related to a problem? Please describe.

It would be awesome if we could automatically set the GOMEMLIMIT for the collector (and TA/Bridge too) when a user sets a limit on the k8s deployment.

### Describe the solution you'd like

when we create the deployment/sts/daemon we could read the resource limit and if present set GOMEMLIMIT to 80% of that per the recommendation [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/extension/ballastextension/README.md)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** 
There's been a lot written about the topic, but basically we should always be setting these for any of the go apps we run.

([good article from John Howard, Istio Maintainer](https://blog.howardjohn.info/posts/gomaxprocs/))

**Link to tracking Issue(s):**

- Resolves #2919 
- Resolves #1456 

**Testing:** n/a

**Documentation:** n/a, though I should probably document this somewhere?

### CODE REVIEW

File: config/crd/kustomizeconfig.yaml [hunk 1]
```diff
@@ -5,12 +5,12 @@ nameReference:
...
+    path: spec/conversion/webhook/clientConfig/service/name
 
 namespace:
 - kind: CustomResourceDefinition
   group: apiextensions.k8s.io
-  path: spec/conversion/webhookClientConfig/service/namespace
+  path: spec/conversion/webhook/clientConfig/service/namespace
   create: false
 
 varReference:
```
Comment (swiatekm-sumo) [inline config/crd/kustomizeconfig.yaml:16]: Why is this changing at all?

Comment (jaronoff97) [reply config/crd/kustomizeconfig.yaml:16]: my kustomize isn't the best... but this is the patch:
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
metadata:
  name: opentelemetrycollectors.opentelemetry.io
spec:
  conversion:
    strategy: Webhook
    webhook:
      clientConfig:
        service:
          namespace: system
          name: opentelemetry-operator-webhook-service
          path: /convert
```

When i tried to install, i got this error:
```
The CustomResourceDefinition "opentelemetrycollectors.opentelemetry.io" is invalid: spec.conversion.webhookClientConfig: Required value: exactly one of url or service is required
```

Comment (swiatekm-sumo) [reply config/crd/kustomizeconfig.yaml:16]: Isn't this file autogenerated by kubebuilder? If not, where did it come from?

Comment (swiatekm-sumo) [reply config/crd/kustomizeconfig.yaml:16]: FYI, the rest of the PR looks good to me, I'm just confused about this specific file.

Comment (jaronoff97) [reply config/crd/kustomizeconfig.yaml:16]: i don't think this is autogenerated... cc @pavolloffay

File: .chloggen/set-gomemlimit.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. collector, target allocator, auto-instrumentation, opamp, github action)
+component: collector, target allocator, opamp
+
+# A brief description of the change. Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Introduces a new feature gate for `operator.golang.flags` to automatically add the environment variables for GOMAXPROCS and GOMEMLIMIT
```
Comment (pavolloffay) [inline .chloggen/set-gomemlimit.yaml:8]: Do we need to put this behind a feature gate?

File: .chloggen/set-gomemlimit.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+# One or more tracking issues related to the change
+issues: [2919, 1456]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (pavolloffay) [inline .chloggen/set-gomemlimit.yaml:16]: maybe explain what values are set?

File: config/crd/kustomizeconfig.yaml (hunk 1 above)
Comment (pavolloffay) [reply config/crd/kustomizeconfig.yaml:16]: it should be auto-generated, I am not sure why it is changed in this PR

Comment (jaronoff97) [reply config/crd/kustomizeconfig.yaml:16]: this was preventing my local instance from starting... the syntax looked wrong to me... i can revert this and maybe i need to update my local kustomize or something

**Review by swiatekm-sumo — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]
I would skip the feature gate and enable it by default

**Pull request merged**